packages/astro/src/actions/index.ts

```diff
@@ -3,12 +3,24 @@ import type { Plugin as VitePlugin } from 'vite';
 import type { AstroIntegration } from '../@types/astro.js';
 import { viteID } from '../core/util.js';
 import { ACTIONS_TYPES_FILE, RESOLVED_VIRTUAL_MODULE_ID, VIRTUAL_MODULE_ID } from './consts.js';
+import { AstroUserError } from '../core/errors/errors.js';
+
+const serverOutputs = ['server', 'hybrid'];
 
 export default function astroActions(): AstroIntegration {
 	return {
 		name: VIRTUAL_MODULE_ID,
 		hooks: {
 			async 'astro:config:setup'(params) {
+				if (!serverOutputs.includes(params.config.output)) {
```

Comment:
You might want to use https://github.com/withastro/astro/blob/461cd42a0652a6675e72619962a1c0cee902af5c/packages/integrations/vercel/src/lib/prerender.ts#L3 for this